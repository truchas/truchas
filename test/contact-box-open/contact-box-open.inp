&OUTPUTS
  output_t = 0.0, 100.0, 150.0
  output_dt = 5.0, 50.0
  output_dt_multiplier = 1
  !short_output_dt_multiplier = 1
  !long_output_dt_multiplier = 5
  !Long_Edit_Bounding_Coords = 0.0, 8.0e-2, 0.00, 8.0e-2, 0.00, 8.0e-2
/

&MESH
  mesh_file = 'split-rotated-cube.exo'
  coordinate_scale_factor = 0.01
  gap_element_blocks = 3
/

&PHYSICS
  fluid_flow = .false.
  heat_transport = .true.
  solid_mechanics = .true.
/

&DIFFUSION_SOLVER
  abs_temp_tol = 0.0
  rel_temp_tol = 5.0e-5
  abs_enthalpy_tol = 0.0
  rel_enthalpy_tol = 5.0e-5
  nlk_tol = 0.05
  nlk_preconditioner = 'hypre_amg'
  pc_amg_cycles = 2
  verbose_stepping = .true.
/

&NUMERICS
  dt_init = 1.0
  dt_grow = 5.0
  dt_max = 5.0
  dt_min = 1.0e-3
  displacement_nonlinear_solution = 'displacement solve'
  contact_distance = 1.0e-7
  contact_norm_trac = 1.0e4
  contact_penalty = 1.0e3
/

&LINEAR_SOLVER
  name = 'displacement precon'
  method = 'none'
  preconditioning_steps = 4
  relaxation_parameter = 1.6
  preconditioning_method = 'tm_ssor'
/

&NONLINEAR_SOLVER 
  name = 'displacement solve'
  method = 'nlk'
  linear_solver_name = 'displacement precon'
  convergence_criterion = 1.0e-12
  NLK_max_vectors = 10
  NLK_vector_tolerance = 0.001
  maximum_iterations = 2000
/

&BODY
  surface_name = 'from mesh file'
  mesh_material_number = 1
  material_number = 1
  temperature = 308.0
/

&BODY
  surface_name = 'from mesh file'
  mesh_material_number = 2
  material_number = 1
  temperature = 308.0
/

&BODY
  surface_name = 'from mesh file'
  mesh_material_number = 3
  material_number = 2
  temperature = 308.0
/

&DS_BOUNDARY_CONDITION
  name = 'ends'
  variable = 'temperature'
  condition = 'dirichlet'
  face_set_ids = 1, 2
  data_function = 'applied temperature profile'
/

&FUNCTION
  name = 'applied temperature profile'
  type = 'smooth step'
  smooth_step_x0 = 0.0
  smooth_step_y0 = 308.0
  smooth_step_x1 = 100.0
  smooth_step_y1 = 288.0
/

&DS_BOUNDARY_CONDITION
  name = 'lateral sides'
  variable = 'temperature'
  condition = 'flux'
  face_set_ids = 3, 4, 5, 6
  data_constant = 0.0
/

&DS_INTERFACE_CONDITION
  name = 'gap interface'
  variable = 'temperature'
  condition = 'HTC'
  face_set_ids = 3
  data_constant = 1.0e4
/

&BC
  bc_name = 'left end -- no normal displacement'
  surface_name = 'from mesh file'
  mesh_surface = 1
  bc_variable = 'displacement'
  bc_type = 'normal-displacement' 
  bc_value = 0.0
/

&BC
  bc_name = 'gap interface'
  surface_name = 'from mesh file'
  mesh_surface = 10
  bc_variable = 'displacement'
  bc_type = 'contact'
  bc_value = 0.0
/

&BC
  bc_name = 'right end -- no normal displacement'
  surface_name = 'from mesh file'
  mesh_surface = 2
  bc_variable = 'displacement'
  bc_type = 'normal-displacement'
  bc_value = 0.0
/

&BC
  bc_name = 'front side'
  surface_name = 'from mesh file'
  mesh_surface = 3
  bc_variable = 'displacement'
  bc_type = 'normal-displacement'
  bc_value = 0.0
/

&BC
  bc_name = 'bottom side'
  surface_name = 'from mesh file'
  mesh_surface = 5
  bc_variable = 'displacement'
  bc_type = 'normal-displacement'
  bc_value = 0.0
/

&BC
  bc_name = 'top side'
  surface_name = 'from mesh file'
  mesh_surface = 6
  bc_variable = 'displacement'
  bc_type = 'normal-displacement'
  bc_value = 0.0
/

define the material properties (aluminum, SI units)
Elastic constants (stresses) in MPa to make test checking easier 
&MATERIAL
  material_number =  1
  priority =  1
  material_name = 'aluminum'
  material_feature = 'background'
  density =  2.70e+03
  immobile = .true.
/
  Lame1_Constants =  5.20e+10
  Lame2_Constants =  2.60e+10

&PHASE
  name = 'aluminum'
  property_name(1) = 'density',       property_constant(1) = 2.7e3
  property_name(2) = 'specific heat', property_constant(2) = 9.0e2
  property_name(3) = 'conductivity',  property_constant(3) = 2.4e2
  property_name(4) = 'TM reference density',     property_constant(4) = 2.7e3
  property_name(5) = 'TM reference temperature', property_constant(5) = 298.0
  property_name(6) = 'TM linear CTE',            property_constant(6) = 2.2e-5
  property_name(7) = 'Lame1', property_constant(7) = 5.20e+10
  property_name(8) = 'Lame2', property_constant(8) = 2.60e+10
/

&MATERIAL_SYSTEM
  name = 'aluminum'
  phases = 'aluminum'
/

&MATERIAL
  material_number = 2
  priority = 2
  material_name = 'gap'
  density =  0.0
  immobile = .true.
/
  Lame1_Constants = 0.0
  Lame2_Constants = 0.0
