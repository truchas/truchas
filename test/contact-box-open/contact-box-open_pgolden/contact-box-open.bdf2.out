
TRAP step      1: T= 0.00000E+00, H= 1.00000E+00, ETAH= 5.00000E-01
  Preconditioner updated at T= 1.00000E+00, ETAH= 5.00000E-01
    1: error= 1.31029E-01
    2: error= 2.32499E-03
  AIN BCE solve succeeded:   2 iterations, error= 2.32499E-03
  Step accepted: no local error control

BDF2 step      2: T= 1.00000E+00, H= 1.00000E+00, ETAH= 6.66667E-01
    1: error= 7.63695E-01
    2: error= 3.32300E-02
  AIN BCE solve succeeded:   2 iterations, error= 3.32300E-02
  Step accepted: no local error control

BDF2 step      3: T= 2.00000E+00, H= 1.00000E+00, ETAH= 6.66667E-01
    1: error= 1.23751E-01
    2: error= 3.51747E-02
  AIN BCE solve succeeded:   2 iterations, error= 3.51747E-02
  Step accepted: no local error control

BDF2 step      4: T= 3.00000E+00, H= 1.00000E+00, ETAH= 6.66667E-01
    1: error= 9.67581E-02
    2: error= 2.65776E-02
  AIN BCE solve succeeded:   2 iterations, error= 2.65776E-02
  Step accepted: perr= 1.21752E-01

BDF2 step      5: T= 4.00000E+00, H= 2.02442E+00, ETAH= 1.21269E+00
    1: error= 1.66804E-01
    2: error= 1.47844E-01
    3: error= 2.68904E-03
  AIN BCE solve succeeded:   3 iterations, error= 2.68904E-03
  Step accepted: perr= 3.12092E-01

BDF2 step      6: T= 6.02442E+00, H= 1.97464E+00, ETAH= 1.32191E+00
    1: error= 1.02656E-01
    2: error= 5.93840E-02
    3: error= 6.40371E-03
  AIN BCE solve succeeded:   3 iterations, error= 6.40371E-03
  Step accepted: perr= 1.35976E-01

BDF2 step      7: T= 7.99905E+00, H= 3.51691E+00, ETAH= 2.14390E+00
  Preconditioner updated at T= 1.15160E+01, ETAH= 2.14390E+00
    1: error= 3.77934E-01
    2: error= 7.12502E-03
  AIN BCE solve succeeded:   2 iterations, error= 7.12502E-03
  Step accepted: perr= 3.77934E-01

BDF2 step      8: T= 1.15160E+01, H= 3.24811E+00, ETAH= 2.19447E+00
    1: error= 5.00724E-01
    2: error= 2.08018E-02
  AIN BCE solve succeeded:   2 iterations, error= 2.08018E-02
  Step accepted: perr= 5.00724E-01

BDF2 step      9: T= 1.47641E+01, H= 3.07891E+00, ETAH= 2.07107E+00
    1: error= 5.00822E-01
    2: error= 1.53672E-02
  AIN BCE solve succeeded:   2 iterations, error= 1.53672E-02
  Step accepted: perr= 5.00822E-01

BDF2 step     10: T= 1.78430E+01, H= 3.20024E+00, ETAH= 2.11984E+00
    1: error= 5.12903E-01
    2: error= 2.08536E-02
  AIN BCE solve succeeded:   2 iterations, error= 2.08536E-02
  Step accepted: perr= 5.13371E-01

BDF2 step     11: T= 2.10432E+01, H= 3.13030E+00, ETAH= 2.09458E+00
    1: error= 4.97539E-01
    2: error= 1.81046E-02
  AIN BCE solve succeeded:   2 iterations, error= 1.81046E-02
  Step accepted: perr= 4.94944E-01

BDF2 step     12: T= 2.41735E+01, H= 3.15745E+00, ETAH= 2.10194E+00
    1: error= 5.01848E-01
    2: error= 1.89063E-02
  AIN BCE solve succeeded:   2 iterations, error= 1.89063E-02
  Step accepted: perr= 4.99981E-01

BDF2 step     13: T= 2.73310E+01, H= 3.15788E+00, ETAH= 2.10520E+00
    1: error= 4.98766E-01
    2: error= 1.91152E-02
  AIN BCE solve succeeded:   2 iterations, error= 1.91152E-02
  Step accepted: perr= 4.97351E-01

BDF2 step     14: T= 3.04888E+01, H= 3.16189E+00, ETAH= 2.10748E+00
    1: error= 5.00554E-01
    2: error= 1.94002E-02
  AIN BCE solve succeeded:   2 iterations, error= 1.94002E-02
  Step accepted: perr= 5.00452E-01

BDF2 step     15: T= 3.36507E+01, H= 3.15843E+00, ETAH= 2.10600E+00
    1: error= 5.01432E-01
    2: error= 1.93164E-02
  AIN BCE solve succeeded:   2 iterations, error= 1.93164E-02
  Step accepted: perr= 5.01432E-01

BDF2 step     16: T= 3.68092E+01, H= 3.15435E+00, ETAH= 2.10335E+00
    1: error= 5.01965E-01
    2: error= 1.91168E-02
  AIN BCE solve succeeded:   2 iterations, error= 1.91168E-02
  Step accepted: perr= 5.01487E-01

BDF2 step     17: T= 3.99635E+01, H= 3.15172E+00, ETAH= 2.10144E+00
    1: error= 5.02466E-01
    2: error= 1.89740E-02
  AIN BCE solve succeeded:   2 iterations, error= 1.89740E-02
  Step accepted: perr= 5.01530E-01

BDF2 step     18: T= 4.31152E+01, H= 3.14840E+00, ETAH= 2.09930E+00
    1: error= 5.02698E-01
    2: error= 1.88050E-02
  AIN BCE solve succeeded:   2 iterations, error= 1.88050E-02
  Step accepted: perr= 5.01561E-01

BDF2 step     19: T= 4.62636E+01, H= 3.14504E+00, ETAH= 2.09707E+00
    1: error= 5.02958E-01
    2: error= 1.86284E-02
  AIN BCE solve succeeded:   2 iterations, error= 1.86284E-02
  Step accepted: perr= 5.01579E-01

BDF2 step     20: T= 4.94087E+01, H= 3.14176E+00, ETAH= 2.09487E+00
    1: error= 5.03243E-01
    2: error= 1.84557E-02
  AIN BCE solve succeeded:   2 iterations, error= 1.84557E-02
  Step accepted: perr= 5.01584E-01

BDF2 step     21: T= 5.25504E+01, H= 3.13844E+00, ETAH= 2.09266E+00
    1: error= 5.03522E-01
    2: error= 1.82809E-02
  AIN BCE solve succeeded:   2 iterations, error= 1.82809E-02
  Step accepted: perr= 5.01578E-01

BDF2 step     22: T= 5.56889E+01, H= 3.13516E+00, ETAH= 2.09047E+00
    1: error= 5.03802E-01
    2: error= 1.81067E-02
  AIN BCE solve succeeded:   2 iterations, error= 1.81067E-02
  Step accepted: perr= 5.01558E-01

BDF2 step     23: T= 5.88240E+01, H= 3.13193E+00, ETAH= 2.08831E+00
    1: error= 5.04077E-01
    2: error= 1.79351E-02
  AIN BCE solve succeeded:   2 iterations, error= 1.79351E-02
  Step accepted: perr= 5.01526E-01

BDF2 step     24: T= 6.19560E+01, H= 3.12879E+00, ETAH= 2.08621E+00
    1: error= 5.04342E-01
    2: error= 1.77669E-02
  AIN BCE solve succeeded:   2 iterations, error= 1.77669E-02
  Step accepted: perr= 5.01481E-01

BDF2 step     25: T= 6.50847E+01, H= 3.12576E+00, ETAH= 2.08418E+00
    1: error= 5.04594E-01
    2: error= 1.76039E-02
  AIN BCE solve succeeded:   2 iterations, error= 1.76039E-02
  Step accepted: perr= 5.01424E-01

BDF2 step     26: T= 6.82105E+01, H= 3.12286E+00, ETAH= 2.08223E+00
    1: error= 5.04832E-01
    2: error= 1.74472E-02
  AIN BCE solve succeeded:   2 iterations, error= 1.74472E-02
  Step accepted: perr= 5.01353E-01

BDF2 step     27: T= 7.13334E+01, H= 3.12013E+00, ETAH= 2.08039E+00
    1: error= 5.05053E-01
    2: error= 1.72984E-02
  AIN BCE solve succeeded:   2 iterations, error= 1.72984E-02
  Step accepted: perr= 5.01270E-01

BDF2 step     28: T= 7.44535E+01, H= 3.11758E+00, ETAH= 2.07867E+00
    1: error= 5.05254E-01
    2: error= 1.71590E-02
  AIN BCE solve succeeded:   2 iterations, error= 1.71590E-02
  Step accepted: perr= 5.01174E-01

BDF2 step     29: T= 7.75711E+01, H= 3.11525E+00, ETAH= 2.07709E+00
    1: error= 5.05433E-01
    2: error= 1.70304E-02
  AIN BCE solve succeeded:   2 iterations, error= 1.70304E-02
  Step accepted: perr= 5.01064E-01

BDF2 step     30: T= 8.06863E+01, H= 3.11316E+00, ETAH= 2.07567E+00
    1: error= 5.05589E-01
    2: error= 1.69141E-02
  AIN BCE solve succeeded:   2 iterations, error= 1.69141E-02
  Step accepted: perr= 5.00942E-01

BDF2 step     31: T= 8.37995E+01, H= 3.11134E+00, ETAH= 2.07443E+00
    1: error= 5.05717E-01
    2: error= 1.68114E-02
  AIN BCE solve succeeded:   2 iterations, error= 1.68114E-02
  Step accepted: perr= 5.00807E-01

BDF2 step     32: T= 8.69108E+01, H= 3.10981E+00, ETAH= 2.07337E+00
    1: error= 5.05817E-01
    2: error= 1.67240E-02
  AIN BCE solve succeeded:   2 iterations, error= 1.67240E-02
  Step accepted: perr= 5.00659E-01

BDF2 step     33: T= 9.00206E+01, H= 3.10860E+00, ETAH= 2.07253E+00
    1: error= 5.05885E-01
    2: error= 1.66531E-02
  AIN BCE solve succeeded:   2 iterations, error= 1.66531E-02
  Step accepted: perr= 5.00499E-01

BDF2 step     34: T= 9.31292E+01, H= 3.10773E+00, ETAH= 2.07192E+00
    1: error= 5.05919E-01
    2: error= 1.66003E-02
  AIN BCE solve succeeded:   2 iterations, error= 1.66003E-02
  Step accepted: perr= 5.00325E-01

BDF2 step     35: T= 9.62370E+01, H= 3.10724E+00, ETAH= 2.07155E+00
    1: error= 5.05917E-01
    2: error= 1.65670E-02
  AIN BCE solve succeeded:   2 iterations, error= 1.65670E-02
  Step accepted: perr= 5.00139E-01

BDF2 step     36: T= 9.93442E+01, H= 3.10715E+00, ETAH= 2.07144E+00
    1: error= 2.04471E+00
    2: error= 3.27730E-02
  AIN BCE solve succeeded:   2 iterations, error= 3.27730E-02
  Step accepted: perr= 2.04471E+00

BDF2 step     37: T= 1.02451E+02, H= 1.31973E+00, ETAH= 1.01665E+00
    1: error= 1.55663E+00
    2: error= 3.93371E-01
    3: error= 2.94003E-02
  AIN BCE solve succeeded:   3 iterations, error= 2.94003E-02
  Step accepted: perr= 1.55663E+00

BDF2 step     38: T= 1.03771E+02, H= 1.07383E+00, ETAH= 7.41273E-01
    1: error= 6.05889E-01
    2: error= 3.18797E-01
    3: error= 1.49088E-02
  AIN BCE solve succeeded:   3 iterations, error= 1.49088E-02
  Step accepted: perr= 3.09630E-01

BDF2 step     39: T= 1.04845E+02, H= 1.84521E+00, ETAH= 1.13056E+00
    1: error= 6.40729E-01
    2: error= 1.93258E-01
    3: error= 1.67576E-03
  AIN BCE solve succeeded:   3 iterations, error= 1.67576E-03
  Step accepted: perr= 4.50120E-01

BDF2 step     40: T= 1.06690E+02, H= 1.61583E+00, ETAH= 1.10156E+00
    1: error= 1.91848E-01
    2: error= 5.61972E-02
    3: error= 1.61691E-03
  AIN BCE solve succeeded:   3 iterations, error= 1.61691E-03
  Step accepted: perr= 1.36014E-01

BDF2 step     41: T= 1.08306E+02, H= 3.05886E+00, ETAH= 1.84899E+00
    1: error= 1.70643E-01
    2: error= 1.01523E-02
  AIN BCE solve succeeded:   2 iterations, error= 1.01523E-02
  Step accepted: perr= 1.60490E-01

BDF2 step     42: T= 1.11365E+02, H= 4.34711E+00, ETAH= 2.73924E+00
    1: error= 2.59899E-02
    2: error= 3.30260E-03
  AIN BCE solve succeeded:   2 iterations, error= 3.30260E-03
  Step accepted: perr= 2.89638E-02

BDF2 step     43: T= 1.15712E+02, H= 5.00000E+00, ETAH= 3.25749E+00
    1: error= 1.77783E-02
    2: error= 3.66773E-03
  AIN BCE solve succeeded:   2 iterations, error= 3.66773E-03
  Step accepted: perr= 2.12301E-02

BDF2 step     44: T= 1.20712E+02, H= 5.00000E+00, ETAH= 3.33333E+00
    1: error= 1.18615E-02
    2: error= 2.76177E-03
  AIN BCE solve succeeded:   2 iterations, error= 2.76177E-03
  Step accepted: perr= 1.44804E-02

BDF2 step     45: T= 1.25712E+02, H= 5.00000E+00, ETAH= 3.33333E+00
    1: error= 3.83890E-03
    2: error= 8.98781E-04
  AIN BCE solve succeeded:   2 iterations, error= 8.98781E-04
  Step accepted: perr= 4.66224E-03

BDF2 step     46: T= 1.30712E+02, H= 5.00000E+00, ETAH= 3.33333E+00
    1: error= 4.61889E-04
  AIN BCE solve succeeded:   1 iterations, error= 4.61889E-04
  Step accepted: perr= 4.61889E-04

BDF2 step     47: T= 1.35712E+02, H= 5.00000E+00, ETAH= 3.33333E+00
    1: error= 3.98048E-05
  AIN BCE solve succeeded:   1 iterations, error= 3.98048E-05
  Step accepted: perr= 3.98048E-05

BDF2 step     48: T= 1.40712E+02, H= 5.00000E+00, ETAH= 3.33333E+00
    1: error= 3.31164E-04
  AIN BCE solve succeeded:   1 iterations, error= 3.31164E-04
  Step accepted: perr= 3.31164E-04

BDF2 step     49: T= 1.45712E+02, H= 5.00000E+00, ETAH= 3.33333E+00
    1: error= 2.11652E-04
  AIN BCE solve succeeded:   1 iterations, error= 2.11652E-04
  Step accepted: perr= 2.11652E-04
